src/main/java/org/intellij/sdk/codesync/HandleBuffer.java

```diff
@@ -0,0 +1,340 @@
+                                System.out.printf("Error while sending the file to the server: %s", diffFile.fileRelativePath);
+                            }
+                        });
+                    } catch (WebSocketConnectionError  error) {
+                        System.out.printf("Failed to connect to websocket endpoint: %s", WEBSOCKET_ENDPOINT);
+                    }
+                } else {
+                    System.out.printf("Failed to connect to websocket endpoint: %s", WEBSOCKET_ENDPOINT);
+                }
+
+            });
+        }
+    }
+
+    public static void handleDirRename(DiffFile diffFile) {
+        System.out.printf("Populating buffer for dir rename to %s", diffFile.newPath);
+
+        try {
+            Stream<Path> files = Files.walk(Paths.get(diffFile.newPath));
+            files.forEach((path) -> {
+                String newFilePath = path.toString();
+                String oldFilePath = newFilePath.replace(diffFile.newPath, diffFile.oldPath);
+                String newRelativePath = newFilePath.replace(String.format("%s/", diffFile.repoPath), "");
+                String oldRelativePath = oldFilePath.replace(String.format("%s/", diffFile.repoPath), "");
+                String branch = Utils.GetGitBranch(diffFile.repoPath);
+
+                JSONObject diff = new JSONObject();
+                diff.put("old_abs_path", oldFilePath);
+                diff.put("new_abs_path", newFilePath);
+                diff.put("old_rel_path", oldRelativePath);
+                diff.put("new_rel_path", newRelativePath);
+                Utils.WriteDiffToYml(
+                    diffFile.repoPath, branch, newRelativePath, diff.toJSONString(),
+                    false, false, true, false
+                );
+            });
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static boolean handleNewFile(CodeSyncClient client, DiffFile diffFile, ConfigFile configFile, ConfigRepo repo, ConfigRepoBranch configRepoBranch) {
+        String branch = Utils.GetGitBranch(repo.repoPath);
+
+        String originalsFilePath = String.format(
+            "%s/%s/%s/%s", ORIGINALS_REPO, repo.repoPath.substring(1), branch, diffFile.fileRelativePath
+        );
+        File originalsFile = new File(originalsFilePath);
+
+        if (!originalsFile.exists()) {
+            System.out.printf("Original file: %s not found", originalsFilePath);
+        }
+
+        if (Utils.shouldIgnoreFile(diffFile.fileRelativePath, repo.repoPath)) {
+            System.out.printf("Ignoring new file upload: %s", diffFile.fileRelativePath);
+            originalsFile.delete();
+            return true;
+        }
+
+        System.out.printf("Uploading new file: %s \n", diffFile.fileRelativePath);
+        try {
+            Integer fileId = client.uploadFile(repo, diffFile, originalsFile);
+            configRepoBranch.updateFileId(diffFile.fileRelativePath, fileId);
```

Comment:
I think it should return updated JSON config to the for..loop so that next diff-files get the file_id of newly created file
Ref: https://github.com/codesyncapp/codesync-vs/blob/master/src/buffer_handler.ts#L166
(I think brew daemon also does the same)

Reply:
Yes, `configRepoBranch.updateFileId` updates the fileId of the `configFile` instance and next line publishes the change to the yml file as well.

Reply:
That's good. Just make sure next diff file in the same loop uses the updated config file